repose-aggregator/commons/utilities/src/test/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapperTest.scala

```diff
@@ -93,6 +94,7 @@ class HttpServletRequestWrapperTest extends FunSpec with BeforeAndAfter with Mat
       wrappedRequest.getHeaderNames.asScala.toList should contain theSameElementsAs headerMap.keys.filterNot( _ == "foo")
     }
 
+    //todo: fix this test -- it should not be case sensitive
```

Comment:
this test is right, your changes are the problem. header names are case insensitive and looks like your changes have broken that, and are returning multiples. plus you've forced us to handle the case insensitivity everywhere else in our code, which someone will forget to do eventually on a change. that's why we made the header name wrapper in the first place.